app/src/main/java/org/sopt/sample/presentation/login/LoginViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package org.sopt.sample.presentation.login
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import org.sopt.sample.data.dto.request.RequestLoginDto
+import org.sopt.sample.data.dto.response.ResponseLoginDto
+import org.sopt.sample.data.local.State
+import org.sopt.sample.data.repository.AuthRepository
+import timber.log.Timber
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val authRepository: AuthRepository
+) : ViewModel() {
+    // Backing Property
+    private val _loginResult = MutableLiveData<ResponseLoginDto>()
+    val loginResult: LiveData<ResponseLoginDto>
+        get() = _loginResult
+
+    private val _stateMessage = MutableLiveData<State>()
+    val stateMessage: LiveData<State>
+        get() = _stateMessage
+
+    val emailText = MutableLiveData("")
+    val pwdText = MutableLiveData("")
+
+    /** 서버에 로그인 요청 */
+    fun login(email: String, password: String) {
+        if (!checkEmail(email)) {
+            _stateMessage.value = State.INCORRECT_EMAIL
+            return
+        }
+
+        if (!checkPwd(password)) {
+            _stateMessage.value = State.INCORRECT_PWD
+            return
+        }
+
+        viewModelScope.launch {
+            authRepository.login(RequestLoginDto(email, password))
+                .onSuccess { response ->
+                    if (response.status in 200..299) {
+                        Timber.d("LOGIN SUCCESS")
+                        Timber.d("response : $response")
+                        _loginResult.value = response
+                        _stateMessage.value = State.SUCCESS
+                    } else {
+                        Timber.e("LOGIN FAIL")
+                        Timber.e("status code : $response.status")
+                        Timber.e("message : $response.message")
+                        _stateMessage.value = State.FAIL
+                    }
+                }
+                .onFailure {
+                    Timber.e("LOGIN SERVER ERROR")
+                    Timber.e("message : " + it.message)
+                    _stateMessage.value = State.SERVER_ERROR
+                }
+        }
+    }
```

Comment:
```suggestion
        viewModelScope.launch {
            authRepository.login(RequestLoginDto(email, password))
                .onSuccess { 
                    _stateMessage.value = State.SUCCESS
                }
                .onFailure {
                    _stateMessage.value = State.SERVER_ERROR
                }
        }
    }
```
if문을 지워도 성공일 때만 `onsuccess` 안에 들어오기 때문에 이런 식으로 코드를 줄일 수 있을 것 같습니다